_posts/2024-11-19-release-notes-3.6.0.md

```diff
@@ -0,0 +1,234 @@
+With this change to the language, you're able to define multiple type parameter lists or place them after the first arguments list. Clause interleaving would benefit the path-dependent API creators.
+It would eliminate the need for intermediate representations introducing runtime overhead or usage of complicated polymorphic functions.
+
+```scala
+trait Key { type Value }
+trait DB {
+  def getOrElse(k: Key)[V >: k.Value](default: V): V // dependent type parameter
+}
+```
+
+## SIP-58 - Named Tuples
+
+Another stabilized feature in this release are the Named Tuples. These have been introduced as experimental in Scala 3.5.0 and allowed you to give meaningful names to tuple elements and use those names during constructing, destructuring, and pattern matching.
+
+```scala
+extension [T](seq: Seq[T])
+  def partitionBy(predicate: PartialFunction[T, Boolean]): (matching: Seq[T], unmatched: Seq[T]) =
+    seq.partition(predicate.unapply(_).isDefined)
+
+@main def onlySmallRealNumbers =
+  List(
+    (x = 1, y = 0),
+    (x = 2, y = 3),
+    (x = 0, y = 1),
+    (x = 3, y = 0),
+  ).partitionBy:
+    case (x = real, y = 0) => real < 5
+  .matching.map(_.x)
+  .foreach(println)
+```
+
+This change also introduces improvements to extractors of case classes. You can now define named extractors for a selection of fields, allowing you to unclutter your code from unused variables.
+
+```scala
+case class User(id: Int, name: String, surname: String)
+
+extension (values: Seq[User])
+  //  Collect user IDs of every entry that has the name matching argument
+  def idsWithName(name: String) = values.collect:
+    case User(name = `name`, id = userId) => userId
+```
+
+Last, but not least, named tuples are opening a new paradigm of metaprogramming by letting you to compute structural types without need for macros!
+The `Selectable` trait now has a `Fields` type member that can be instantiated to a named tuple.
+
+```scala
+class QueryResult[T](rawValues: Map[String, Any]) extends Selectable:
+  type Fields = NamedTuple.Map[NamedTuple.From[T], Option]
+  def selectDynamic(fieldName: String) = rawValues.get(fieldName)
+  
+case class City(zipCode: Int, name: String)
+
+@main def Test =
+  val query: QueryResult[City] = QueryResult(Map("name" -> "Lausanne"))
+  assert(query.name.contains("Lausanne"))
+  assert(query.zipCode.isEmpty)
+```
+
+You can read more about named tuples in the [dedicated section of Scala 3 reference documentation](https://scala-lang.org/api/3.6.0/docs/docs/reference/other-new-features/named-tuples.html).
+
+## SIP-62 - For-Comprehension Improvements
+
+Starting with Scala 3.6.0 you can take advantage of improvements to the for-comprehesnions syntax.
+Major user-facing improvement introduced by [SIP-62](https://docs.scala-lang.org/sips/better-fors.html) is ability to start for-comprehension block with aliases:
+
+```scala
+for {
+  a = 1
+  b <- Some(2)
+  c <- doSth(a)
+  extension (values: Seq[T]) def toSorted: Seq[T] = ???
```

Comment:
This indentation can't be right, here. I don't know what the snippet is supposed to contain.